src/app/market/sell/add-item/add-item.component.ts

```diff
@@ -354,12 +356,16 @@ export class AddItemComponent implements OnInit, OnDestroy {
     this.upsert().then(t => {
       this.modals.unlock({timeout: 30}, (status) => {
         this.template.post(t, 1).toPromise().then(listing => {
-        this.snackbar.open('Succesfully added Listing!')
-        console.log(listing);
-        this.backToSell();
+          this.isInProcess = false;
+          this.snackbar.open('Succesfully added Listing!')
+          console.log(listing);
+          this.backToSell();
         });
       });
-    }, err => this.snackbar.open(err));
+    }, err => {
+      this.isInProcess = false;
+      this.snackbar.open(err)
```

Comment:
Do we not want to rather output a more user friendly error message here if this fails?

Reply:
```
https://github.com/particl/particl-desktop/blob/market/src/app/core/market/market.service.ts#L46
```
We have already handled the error in the service regarding the message and if backend return differently then at least we have that error to return in the same way as what it is.

Reply:
I agree, having the error is helpful. But if we could potentially be printing errors that include line numbers, indications of things being undefined, etc, then those types of error messages may not be helpful to a user (non-developer user type of person). If the error message is vague, it does not indicate what actually did go wrong, for example: 
was the listing successfully published but there was a problem changing the view, OR
did the publish fail because they had some invalid value, OR
did the publish fail because of a connectivity issue, 
etc

Perhaps prefixing the error message with something indicating that the listing failed to be published would be a little more helpful to the user than just presenting an error message.In other words, an error message needs to tell the user what went wrong but at the same time hint at what action they should try to perform next. In this case, a 'Failed to publish your listing' type of error suggests that the subsequent action may be to try republish the listing again (it is explicitly known that the previous attempt failed, rather than the user needing to first verify whether it possibly succeeded or not).

Reply:
@zaSmilingIdiot  Now your call on the same?